src/App.tsx

```diff
@@ -26,70 +26,103 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+  let visibleGoods = [...goods];
+  const { ALPHABET, LENGTH, NONE } = SortType;
+
+  switch (sortType) {
+    case LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    case ALPHABET:
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case NONE:
+      visibleGoods = [...goods];
+      break;
+    default:
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
-  // Sort and reverse goods if needed
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
-
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
 };
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: 0,
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+
+    const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+    const { ALPHABET, LENGTH, NONE } = SortType;
+
+    return (
+      <>
+        <div className="section content">
+          <div className="buttons">
+            <button
+              onClick={() => this.setState({ sortType: ALPHABET })}
+              type="button"
+              className={`button is-info ${sortType === ALPHABET ? '' : 'is-light'}`}
+            >
+              Sort alphabetically
+            </button>
+
+            <button
+              type="button"
+              className={`button is-success ${sortType === LENGTH ? '' : 'is-light'}`}
```

Comment:
You can use some libs (for example `classnames`) to use conditional class names. It may be more readable.

Reply:
my personal opinion is that it is better and clearer
I know how to use this library